### DISCUSSION TITLE
var-naming[no-role-prefix] struggles

### DISCUSSION THREAD

**Discussion by kaoscoach** [ansible/ansible-lint#4214] [Q&A]
I have a role called king_install. In the king_install directory of the role, I have a vars folder with a main.yml file. main.yml, has one line,

king_install_user: ansible_user

I use this variable to populate a file with ansible.builtin.copy and a content: directive as such

export USER={{ king_install_user }}

The playbook works as intended. However when I run ansible-lint to check my syntax, I get the 'error' var-naming[no-role-prefix'. I have no idea how to resolve this. I have renamed the variable with the king_install_ prefix so I am not sure what it is flagging. Can someone help?

**Comment by cavcrosby** [comment]
What version of `ansible-lint` are you using? Also, could you write down the exact steps to reproduce the problem?
